src/api/repositories/user/user_repository_mock.go

```diff
@@ -0,0 +1,35 @@
+package repositories
+
+import (
+	"github.com/fullstacktf/personal-nutritionist-backend/api/models"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/mock"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+type UserRepositoryMock struct {
+	mock.Mock
+}
+
+func (m *UserRepositoryMock) GetUsers(c *gin.Context) ([]models.User, error) {
+	args := m.Called(c)
+	return args.Get(0).([]models.User), nil
+}
+
+func (m *UserRepositoryMock) GetUserByID(c *gin.Context, id primitive.ObjectID) (models.User, error) {
+	var user models.User
+	return user, nil
```

Comment:
I think we return a null user.

I believe we should implement the mock methods like:
 ```go
 args := m.Called(c) 
 return args.Get(0).(models.User), args.Get(1)
 ```
 
 Step 1 get args from context.
 Step 2 return the arg 0 and arg 1

Reply:
Same in the rest of the methods